frontend/src/core/editor/page/views/editorComponentListView.js

```diff
@@ -77,30 +77,36 @@ define(function(require) {
     renderComponentList: function() {
       Origin.trigger('editorComponentListView:removeSubviews');
       // _.each(this.collection, function(componentType) {
+
       this.collection.each(function(componentType) {
         var properties = componentType.get('properties');
-        if (properties && properties.hasOwnProperty('._supportedLayout')) {
-          var supportedLayout = properties.hasOwnProperty('._supportedLayout').enum;
+        if (properties && properties.hasOwnProperty('_supportedLayout')) {
+          var supportedLayout = properties._supportedLayout.enum;
+          var availablePositions = {
```

Comment:
@danielstorey I think availablePositions is also used to find positions that are not already occupied as set in setupFilters function. Assigning a new value to availablePositions here allows the user to insert two left (or right) positioned components.